pam/pam_adsys.c

```diff
@@ -223,14 +262,36 @@ static int set_dconf_profile(pam_handle_t * pamh, const char *username,
 {
 	int retval = PAM_SUCCESS;
 
+	char* profile_name = (char*)malloc((strlen(username) + 1) * sizeof(char));
+	strcpy(profile_name, username);
+
+	// We need to check if the profile name does not already contain the domain or is in the \\format
+	if (strchr(profile_name, '@') == NULL && strchr(profile_name, '\\') == NULL) {
```

Comment:
So if someone is using the syntax `domain\\user`, what is then the DCONF_PROFILE set to?

Reply:
I don't know how pam handles this kind of username, but if it doesn't change it then it would be set to `\\domain\user`, so I guess the parsing should be a bit more robust

Reply:
I think you really should give it a try and see how it goes. If it doesn’t replace with @, then, we will have again the issue with DCONF_PROFILE for this use case.